### ISSUE TITLE
Adding a function to replace strings in the textreplacer

### ISSUE THREAD

**Issue by Foorcee** [KyoriPowered/adventure#236]
Hello,
I would find it very handy if you had the possibility to replace strings in the text replacer as well, e.g. the action in click events. With hover events it already works. Currently I have to iterate over all components again and replace the action if necessary.

I would be very happy if you could find a solution for this.

**Comment by Phoenix616** [comment]
What's the general opinion on how this should be implemented and exposed in the API? Seeing as click events don't use components the hover event logic in the TextReplacementRenderer can't be used in a similar way and a separate method specific to the click events would be necessary.

Also should this still be part of the TextReplacementRenderer at all or completely separate? I personally would kinda expect it to still be in that API as well as the Component#replaceText utility method (that's how I noticed this shortcoming initially) but maybe there are better ideas/differing opinions on that.

**Comment by zml2008** [comment]
In about 5 minutes, I've done up a quick implementation of the plain-string replacement using existing API:

```java
          Component.textOfChildren(component).toBuilder().mapChildrenDeep(x -> {
            if (x.clickEvent() != null) {
              return (BuildableComponent<?, ?>) x.clickEvent(ClickEvent.clickEvent(x.clickEvent().action(), /* replaced */ x.clickEvent().value()));
            } else {
              return x;
            }
          });
```

There are some nasty casts that hurt ergonomics here (which can hopefully be resolved for 5.0.0), but imo since there really isn't the same complexity that exists with replacing within styled content, it's not worth trying to shove into the existing replacement systems.

If someone refines an implementation of click event replacement, we might accept some elements as a PR, but please talk to us before spending much time on that.
